apps/client/src/components/app/table-user.tsx

```diff
@@ -0,0 +1,153 @@
+        end={
+          <Button>
+            <Plus />
+            Add user
+          </Button>
+        }
+      />
+
+      <Table>
+        <TableHeader>
+          <TableRow>
+            <TableHead>User</TableHead>
+            <TableHead>Role</TableHead>
+            <TableHead>Joined</TableHead>
+            <TableHead>
+              <span className="sr-only">User modification options</span>
+            </TableHead>
+          </TableRow>
+        </TableHeader>
+        <TableBody>
+          {isLoading ? (
+            <TableLoader rowsOnly />
+          ) : !users || users.length === 0 ? (
+            <TableRow>
+              <TableCell
+                colSpan={4}
+                className="text-center py-8 text-muted-foreground"
+              >
+                {search ? `No users found for "${search}"` : "No users found"}
+              </TableCell>
+            </TableRow>
+          ) : (
+            users.map((user) => (
+              <TableRow key={user.id}>
+                <TableCell>
+                  <span className="flex items-center gap-2">
+                    <Avatar>
+                      <AvatarImage src={user.photo} />
+                      <AvatarFallback>{formatInitials(user)}</AvatarFallback>
+                    </Avatar>
+                    {formatFullName(user)}
+                  </span>
+                </TableCell>
+                <TableCell>{getRoleById(user.roleId, roles || [])}</TableCell>
+                <TableCell>{formatDate(user.createdAt)}</TableCell>
+                <TableCell className="text-right">
+                  <DropdownMenu>
+                    <DropdownMenuTrigger asChild>
+                      <Button variant="ghost" size="icon" className="size-6">
+                        <MoreHorizontal />
+                      </Button>
+                    </DropdownMenuTrigger>
+                    <DropdownMenuContent align="end">
+                      <DropdownMenuItem
+                        onClick={() => {
+                          window.alert("not implemented");
+                        }}
+                      >
+                        Edit user
+                      </DropdownMenuItem>
+                      <DropdownMenuItem
+                        onClick={() => {
+                          setUserToDelete(user);
+                          setIsDialogOpen(true);
+                        }}
+                      >
+                        Delete user
+                      </DropdownMenuItem>
+                    </DropdownMenuContent>
+                  </DropdownMenu>
+                </TableCell>
+              </TableRow>
+            ))
+          )}
+        </TableBody>
+      </Table>
+      <DialogDeleteUser
+        user={
+          userToDelete ||
+          ({
+            id: "",
+            first: "",
+            last: "",
+            roleId: "",
+            photo: "",
+          } as User)
+        }
```

Comment:
This provides a safe fallback with empty strings to prevent runtime errors, but I feel like theres a cleaner way to do this. With more time I would have looked in to it a bit more...